### PR TITLE
resolved issue where proper github id not being used

### LINKED ISSUE #7: Not using proper commit hash

### PR DESCRIPTION
**Pull request by hjaensch7**
Resolves #7

### CODE REVIEW

**Comment by hjaensch7** [comment]
- [ ] Add queuing for race condition
- [ ] Start all repos added when handling installation
- [x] Only deploy from default branch
- [x] Add actual commit hash

File: pushHandler.go [hunk 1]
```diff
@@ -40,32 +40,57 @@ func (h *pushHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+	clonedPath, err := cloneRepo(repoFullName)
+	if err != nil {
+		log.Error.Println("Error cloning repo:", err)
+		return
+	}
+	defer os.RemoveAll(clonedPath) // clean up
+
+	//If first time installation get hash
+	if hash == "" {
+		//wd, _ := os.Getwd()
```
Comment (Others) [inline pushHandler.go]: Remove commented code

File: pushHandler.go [hunk 2]
```diff
@@ -40,32 +40,57 @@ func (h *pushHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+	// Get Repo Contents
+	log.Info.Println("Cloning Repo...")
+	clonedPath, err := cloneRepo(repoFullName)
+	if err != nil {
+		log.Error.Println("Error cloning repo:", err)
+		return
+	}
+	defer os.RemoveAll(clonedPath) // clean up
+
+	//If first time installation get hash
```
Comment (Others) [inline pushHandler.go]: Could be phrased more clearly

File: pushHandler.go (hunk 1 above)
Comment (Others) [inline pushHandler.go]: Remove commented code

File: cloneRepo.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+func getHash(repoFilePathAbs string) (hash string, err error) {
+	cmd := exec.Command("git", "rev-parse", "HEAD")
+	cmd.Dir = repoFilePathAbs
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err = cmd.Run()
+	hash = stdout.String()
+	hash = hash[:len(hash)-1]
```
Comment (Others) [inline cloneRepo.go:44]: If no input does this give a overwrite?

File: cloneRepo.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+	"gopkg.in/src-d/go-git.v4"
+)
+
+//Clone repo into temp dir
+func cloneRepo(repoName string) (dirName string, err error) {
+	// Tempdir to clone the repository
+	dir, err := ioutil.TempDir("", ".git_")
+	if err != nil {
+		log.Error.Println(err)
```
Comment (Others) [inline cloneRepo.go:18]: return `err`?

File: cloneRepo.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	}
+
+	_, err = git.PlainClone(dir, false, &git.CloneOptions{
+		URL: "https://github.com/" + repoName + ".git",
+	})
+
+	if err != nil {
+		log.Error.Println(err)
```
Comment (Others) [inline cloneRepo.go:27]: return `err`?

**Review by Others — CHANGES_REQUESTED** [review]

**Review by Others — APPROVED** [review]

**Pull request merged**